### PR TITLE
Add help for all internal options valid for default roles.

### PR DESCRIPTION
**Pull request by dwsteele**
Fix the segfault when help for an internal option is requested by adding help for all internal options that are valid for a default command role.

Also print warnings about internal options in code rather than putting in each command/option description.

This is the rest of the fix for https://github.com/pgbackrest/pgbackrest/projects/2#card-51636667.

### CODE REVIEW

File: doc/lib/pgBackRestDoc/Common/DocConfig.pm [hunk 1]
```diff
@@ -286,11 +280,14 @@ sub process
...
-
-                    if (!defined($strSection))
+                    if (defined($oOptionDoc))
                     {
-                        $strSection = "general";
+                        $strSection = $oOptionDoc->paramGet('section', false);
+
+                        if (!defined($strSection))
+                        {
+                            $strSection = "general";
```
Comment (pgstef) [inline doc/lib/pgBackRestDoc/Common/DocConfig.pm:289]: Shouldn't this be inlined without brackets?
```
if (!defined($strSection))
    $strSection = "general";
```

**Comment by pgstef** [comment]
Tried it:
```bash
$ pgbackrest version
pgBackRest 2.33
$ pgbackrest help backup repo
#--OK--
$ pgbackrest help backup repo-local
Segmentation fault
$ pgbackrest help backup repo-path
#--OK--
$ pgbackrest help backup repo1-path
#--OK--

$ pgbackrest version
pgBackRest 2.34dev
$ pgbackrest help backup repo
#--OK--
$ pgbackrest help backup repo-local
pgBackRest 2.34dev - 'backup' command - 'repo-local' option help

Repository is local.

Indicates that the repository for the specified index is local and any
repository host settings (e.g. repo1-host) should be ignored.

FOR INTERNAL USE ONLY. DO NOT USE IN PRODUCTION.

default: n
$ pgbackrest help backup repo-path
#--OK--
$ pgbackrest help backup repo1-path
#--OK--
```

It seems to work as expected and looks good to me then.

File: doc/lib/pgBackRestDoc/Common/DocConfig.pm (hunk 1 above)
Comment (dwsteele) [reply doc/lib/pgBackRestDoc/Common/DocConfig.pm:289]: I wish. Perl requires the braces even for a single line.

**Comment by dwsteele** [comment]
> It seems to work as expected and looks good to me then.

Excellent. If you are done with your review please approve the PR. Thank you!

File: doc/lib/pgBackRestDoc/Common/DocConfig.pm (hunk 1 above)
Comment (pgstef) [reply doc/lib/pgBackRestDoc/Common/DocConfig.pm:289]: Oh right, it's a perl file :man_facepalming:

**Review by pgstef — APPROVED** [review]

File: doc/lib/pgBackRestDoc/Common/DocConfig.pm (hunk 1 above)
Comment (dwsteele) [reply doc/lib/pgBackRestDoc/Common/DocConfig.pm:289]: Not too many Perl mods these days, luckily!

File: doc/xml/reference.xml [hunk 2]
```diff
@@ -210,6 +210,24 @@
...
+                        <text>The number of retries for jobs run in the local process, e.g. file backup.</text>
+
+                        <example>5</example>
+                    </config-key>
+
+                    <!-- ======================================================================================================= -->
+                    <config-key id="job-retry-interval" name="Job Retry Interval">
+                        <summary>Retry interval after first retry.</summary>
+
+                        <text>The first retry is always performed immediately. This options specifies the interval to wait (in seconds) before performing each additional retry.</text>
```
Comment (cmwshang) [inline doc/xml/reference.xml:226]: `This options` should not be plural: `This option`

File: doc/xml/reference.xml [hunk 3]
```diff
@@ -488,6 +515,15 @@
                         <example>y</example>
                     </config-key>
 
+                    <!-- ======================================================================================================= -->
+                    <config-key id="repo-local" name="Repository Locality">
+                        <summary>Repository is local.</summary>
+
+                        <text>Indicates that the repository for the specified index is local and any repository host settings (e.g. <setting>repo1-host</setting>) should be ignored.</text>
```
Comment (cmwshang) [inline doc/xml/reference.xml:522]: Maybe `and any repository host settings for that index` - just to ensure people don't think repo host settings for all repos are ignored.

File: doc/xml/reference.xml [hunk 4]
```diff
@@ -1096,6 +1132,15 @@
                         <example>backupdb</example>
                     </config-key>
 
+                    <!-- ======================================================================================================= -->
+                    <config-key id="pg-local" name="PostgreSQL Locality">
+                        <summary><postgres/> is local.</summary>
+
+                        <text>Indicates that <postgres/> for the specified index is local and any <postgres/> host settings (e.g. <setting>pg1-host</setting>) should be ignored.</text>
```
Comment (cmwshang) [inline doc/xml/reference.xml:1139]: Same comment as for repo-local.

**Review by cmwshang — CHANGES_REQUESTED** [review]
Some typos I will fix and indicate the commit in the conversations.

File: doc/xml/reference.xml (hunk 2 above)
Comment (cmwshang) [reply doc/xml/reference.xml]: Fixed in commit 2023430cbc97bb5deaf0f28372ecaebc1bbce3f7

File: doc/xml/reference.xml (hunk 3 above)
Comment (cmwshang) [reply doc/xml/reference.xml]: Fixed in commit 2023430cbc97bb5deaf0f28372ecaebc1bbce3f7

File: doc/xml/reference.xml (hunk 4 above)
Comment (cmwshang) [reply doc/xml/reference.xml]: Fixed in commit 2023430cbc97bb5deaf0f28372ecaebc1bbce3f7

File: doc/lib/pgBackRestDoc/Common/DocConfig.pm [hunk 5]
```diff
@@ -321,6 +318,8 @@ sub process
             $$oCommandOption{&CONFIG_HELP_SUMMARY} = $oOptionDoc->nodeGet('summary')->textGet();
             $$oCommandOption{&CONFIG_HELP_DESCRIPTION} = $oOptionDoc->textGet();
             $$oCommandOption{&CONFIG_HELP_EXAMPLE} = $oOptionDoc->fieldGet('example');
+            $oCommandOption->{&CONFIG_HELP_INTERNAL} =
+                cfgDefineCommand()->{$strCommand}{&CFGDEF_INTERNAL} ? true : $oOptionDefine->{$strOption}{&CFGDEF_INTERNAL};
```
[the remark this answers is not in the document]
Comment (dwsteele) [reply doc/lib/pgBackRestDoc/Common/DocConfig.pm:322]: That issue was fixed in 45f83558ea862c6f4 and is not part of his commit. But we can pull it in if you want.

File: doc/xml/reference.xml (hunk 2 above)
Comment (dwsteele) [reply doc/xml/reference.xml]: Looks good to me.

File: doc/xml/reference.xml (hunk 3 above)
Comment (dwsteele) [reply doc/xml/reference.xml]: Looks good to me.

File: doc/xml/reference.xml (hunk 4 above)
Comment (dwsteele) [reply doc/xml/reference.xml]: Looks good to me.

File: doc/lib/pgBackRestDoc/Common/DocConfig.pm (hunk 5 above)
[the remark this answers is not in the document]
Comment (dwsteele) [reply doc/lib/pgBackRestDoc/Common/DocConfig.pm:322]: I decided it was a good idea to pull in master so we can verify all these are really fixed. They work as expected now, or at least as I expect them to work!

[the remark this answers is not in the document]
Comment (cmwshang) [reply doc/lib/pgBackRestDoc/Common/DocConfig.pm:322]: Yes, it looks like it is working now.

**Review by cmwshang — APPROVED** [review]

**Pull request merged**